src/interf.jl

```diff
@@ -10,7 +10,9 @@ function nnmf(X::AbstractMatrix{T}, k::Integer;
               W0::Union{AbstractMatrix{T}, Nothing}=nothing,
               H0::Union{AbstractMatrix{T}, Nothing}=nothing,
               update_H::Bool=true,
-              verbose::Bool=false) where T
+              verbose::Bool=false,
+              M = ones(size(X))) where T
```

Comment:
This will lead to unnecessary allocation for most algorithms. A better default would be `M=1` or `M=nothing`. You could use `M::Union{...}` if you want to constrain the types allowed.

Reply:
I will fix this